source/patterns/@aws-solutions-constructs/resources/kms-key-policy-updater/index.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import * as cdk from "aws-cdk-lib";
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import { Aws, CustomResource } from 'aws-cdk-lib';
+import { Provider } from "aws-cdk-lib/custom-resources";
+import { buildLambdaFunction } from "@aws-solutions-constructs/core";
+import { IKey } from "aws-cdk-lib/aws-kms";
+import { Distribution } from "aws-cdk-lib/aws-cloudfront";
+// Note: To ensure CDKv2 compatibility, keep the import statement for Construct separate
+import { Construct } from 'constructs';
+
+export interface CreateKeyPolicyUpdaterResponse {
+  readonly lambdaFunction: lambda.Function;
+  readonly customResource: CustomResource;
+}
+
+export interface KeyPolicyUpdaterProps {
+  readonly encryptionKey: IKey;
+  readonly distribution: Distribution;
+  readonly timeout?: cdk.Duration;
+  readonly memorySize?: number;
+}
+
+export function createKeyPolicyUpdaterCustomResource(
+  scope: Construct,
+  props: KeyPolicyUpdaterProps
+): CreateKeyPolicyUpdaterResponse {
+
+  const lambdaFunction = buildLambdaFunction(scope, {
+    lambdaFunctionProps: {
+      runtime: lambda.Runtime.NODEJS_18_X,
+      handler: 'index.handler',
+      description: 'kms-key-policy-updater',
+      timeout: props.timeout,
+      memorySize: props.memorySize,
+      code: lambda.Code.fromAsset(`${__dirname}/custom-resource/`),
+      role: new iam.Role(scope, 'KmsKeyPolicyUpdateLambdaRole', {
+        assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+        description: 'Role to update kms key policy to allow cloudfront access',
+        inlinePolicies: {
+          KmsPolicy: new iam.PolicyDocument({
+            statements: [
+              new iam.PolicyStatement({
+                actions: ['kms:PutKeyPolicy', 'kms:GetKeyPolicy', 'kms:DescribeKey'],
+                effect: iam.Effect.ALLOW,
+                resources: [ props.encryptionKey.keyArn ]
+              })
+            ]
+          }),
+          CWLogsPolicy: new iam.PolicyDocument({
```

Comment:
I believe that buildLambdaFunction already adds this policy

Reply:
You are correct, fixed this in e054453fcd793d96891eb998c38835264a16097a